app/controllers/openstax/api/v1/api_controller.rb

```diff
@@ -46,8 +48,12 @@ def session_user?
           !!current_session_user && doorkeeper_token.blank?
         end
 
+        def set_date_header
+          response.date = Time.now unless response.date?
```

Comment:
Rails actually does set it here: https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/conditional_get.rb#L209

But only if you call `expires_in` in the controller.